src/nbl/ui/CWindowX11.cpp

```diff
@@ -321,31 +130,178 @@ void CWindowX11::processEvent(XEvent event)
     }
 }
 
-void CWindowManagerX11::init()
+void CWindowManagerX11::CThreadHandler::init()
 {
     //TODO: init x11/xinput loaders here
     display = x11.pXOpenDisplay(nullptr);
 }
 
-void CWindowManagerX11::process_request(SRequest& req)
+void CWindowManagerX11::CThreadHandler::process_request(SRequest& req)
 {
     switch (req.type)
     {
     case ERT_CREATE_WINDOW:
     {
+        // XInitThreads() call not needed unless windows are created concurrently, spoof EGL synchronizes per-display access itself
```

Comment:
just call `XInitThreads` somewhere at the beginning (preferably right after loading the X11 .so) thre's no harm from it

Reply:
its safe to call `XInitThreads` multiple times but not concurrently, so mutex it and put it in the `CX11WindowManager` then you're fine